src/ascii_str.rs

```diff
@@ -389,6 +416,97 @@ impl AsciiExt for AsciiStr {
 }
 
 
+impl<'a> IntoIterator for &'a AsciiStr {
+    type Item = &'a AsciiChar;
+    type IntoIter = Chars<'a>;
+    #[inline]
+    fn into_iter(self) -> Self::IntoIter {
+        self.chars()
+    }
+}
+
+impl<'a> IntoIterator for &'a mut AsciiStr {
+    type Item = &'a mut AsciiChar;
+    type IntoIter = CharsMut<'a>;
+    #[inline]
+    fn into_iter(self) -> Self::IntoIter {
+        self.chars_mut()
+    }
+}
+
+/// An immutable iterator over the characters of an `AsciiStr`.
+pub type Chars<'a> = Iter<'a, AsciiChar>;
+
+/// A mutable iterator over the characters of an `AsciiStr`.
+pub type CharsMut<'a> = IterMut<'a, AsciiChar>;
+
+/// An iterator over the lines of the internal character array.
+#[derive(Debug)]
+pub struct Lines<'a> {
+    // TODO: should this use `core::slice::Split` internally?
```

Comment:
`core::slice::Split` only allows a single byte at a time to be checked. We would check for `\n` and then see if there was a `\r` right before I guess. In this case I think using the `memchr` crate would give a nice speedup.

Do you like to try implementing this? If not, no problem, I'll just open an issue to not forget it.